src/pages/StartingPage.js

```diff
@@ -6,8 +6,47 @@ import Banner from '../components/Banner.js'
 import PageTitle from '../components/PageTitle.js'
 import NameInputBox from "../components/NameInputBox";
 
+import * as Pet from '../data/Pet';
+
 class StartingPage extends Component {
 
+    constructor(props) {
+        super(props);
+        this.state = {
+            selectedPet: null,
+            petName: '',
+        };
+
+        // ensure this is bound correctly after passing to child component
+        this.petTypeHandler = this.petTypeHandler.bind(this);
+        this.petNameHandler = this.petNameHandler.bind(this);
+        this.submitHandler = this.submitHandler.bind(this);
+        this.getState = this.getState.bind(this);
+    }
+
+    petTypeHandler(val) {
+        this.setState({
+            selectedPet: val
+        });
+    }
+
+    petNameHandler(event) {
+        this.setState({
+            petName: event.target.value
+        });
+    }
+
+    getState() {
+        return this.state;
+    }
+
+    submitHandler() {
+        const adoptedPet = Pet.adoptPet(this.state.selectedPet, this.state.petName);
+        this.props.history.push({
+            pathname: '/game',
+            state: { pet: adoptedPet },
+        });
+    }
 
     render() {
         return ([
```

Comment:
remove the `[` and `]` surrounding the return statement and that last key error will go away

Reply:
Yeah this was here before, was unsure if it had a functional reason, but didn't try that